File: UniversalMatrix.cpp
#include<iostream>
#include<string>
using namespace std;
int main()
{
    int arr[60][60],arr2[60][60],arr3[60][60];
    int a,b,i,j;
    char mat;
    int k;
    cout<<"\n----------------------------------------------Welcome in the Universal of 2D Matrix--------------------------------------------------\t"<<endl;
    cout<<endl<<"Enter the Row: ";
    fflush(stdin);
    cin>>a;
    cout<<endl<<"Enter the Coloumn: ";
    fflush(stdin);
    cin>>b;
    cout<<endl<<"Enter Your's Elements in The 1st Matrix:- "<<endl;
    for(i=0;i<a;i++)
     {
     for(j=0;j<b;j++)
         {
            cout<<endl<<"Enter the Element in Mat["<<i<<"]"<<"["<<j<<"]: ";
            fflush(stdin);
            cin>>arr[i][j];
         }
     }
     cout<<endl<<"Enter Your's Elements in The 2nd Matrix:- "<<endl;
     for(i=0;i<a;i++)
      {
        for(j=0;j<b;j++)
         {
          cout<<endl<<"Enter the Element in Mat["<<i<<"]"<<"["<<j<<"]: ";
          fflush(stdin);
          cin>>arr2[i][j];
         }
      }
     cout<<endl<<"Here is Your 1st Matrix:- \n\n";
     for(i=0;i<a;i++)
      {
         for(j=0;j<b;j++)
        {
          cout<<"\t "<<arr[i][j];
        }
          cout<<"\n";
      }
     cout<<endl<<"Here is Your 2nd Matrix:- \n\n";
    for(i=0;i<a;i++)
     {
      for(j=0;j<b;j++)
       {
        cout<<"\t "<<arr2[i][j];
        }
        cout<<"\n";
     }
    cout<<endl<<"Please Choose Your Matrix Operation From there: \n"<<endl;
    cout<<"1.Addition Matrix: A\n2.Subtraction Matrix: S\n3.Multiplication Matrix: M\n4.Division Matrix: D\n5.Transpose Matrix: T\n6.Upper Triangular Matrix: U\n7.Lower Triangular Matrix: L\n8.Diagonals Matrix: X\n9.Tri-Diagonal Matrix: O\n10.Null Matrix: N\n";
    cout<<endl<<"Enter The Matrix Code: ";
    cin>>mat;
    switch(mat)
    {
       case 'A':  cout<<endl<<"Welcome in The Addition of Matrix"<<endl;
                  cout<<endl<<"Here is your Addition Matrix:- \n\n";
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        arr3[i][j]=arr[i][j]+arr2[i][j];
                    }
                  }
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        cout<<"\t "<<arr3[i][j];
                    }
                    cout<<"\n";
                  }
                  cout<<"\n";
                  break;

      case 'S':   cout<<endl<<"Welcome in The Subtraction of Matrix"<<endl;
                  cout<<endl<<"Here is your Subtraction Matrix:- \n\n";
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        arr3[i][j]=arr[i][j]-arr2[i][j];
                    }
                  }
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        cout<<"\t "<<arr3[i][j];
                    }
                    cout<<"\n";
                  }
                  cout<<"\n";
                  break;

      case 'M':   cout<<endl<<"Welcome in The Multiplication of Matrix"<<endl;
                  cout<<endl<<"Here is your Multiplication Matrix:- \n\n";
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                      for(k=0;k<b;k++)
                      {
                         arr3[i][j]=arr3[i][j]+arr[i][k]*arr2[k][j];
                      }
                    }
                  }
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        cout<<"\t "<<arr3[i][j];
                    }
                    cout<<"\n";
                  }
                  cout<<"\n";
                  break;

      case 'D':   cout<<endl<<"Welcome in The Division of Matrix"<<endl;
                  cout<<endl<<"Here is your Division Matrix:- \n\n";
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        arr3[i][j]=arr[i][j]/arr2[i][j];
                    }
                  }
                  for(i=0;i<a;i++)
                  {
                    for(j=0;j<b;j++)
                    {
                        cout<<"\t "<<arr3[i][j];
                    }
                    cout<<"\n";
                  }
                  cout<<"\n";
                  break;

      case 'T':   cout<<endl<<"Welcome in The Transpose of Matrix"<<endl;
                  char choose;
                  cout<<endl<<"If you want to Print Only 1st Matrix Then Type (1)"<<endl<<"If you want to Print Only 2nd Matrix Then Type (2)"<<endl<<"If you want to Print Both Matrixes Then Type (3)";
                  cout<<endl<<"\nEnter Your Choice: ";
                  cin>>choose;
                  switch(choose)
                  {
                    case '1':  cout<<endl<<"Here is Your Transpose of The 1st Matrix:- \n\n";
                                for(i=0;i<b;i++)
                                {
                                  for(j=0;j<a;j++)
                                  {
                                  cout<<"\t "<<arr[j][i];
                                  }
                                  cout<<"\n";
                                }
                                cout<<"\n";
                                break;

                    case '2':   cout<<endl<<"Here is Your Transpose of The 2nd Matrix:- \n\n";
                                for(i=0;i<b;i++)
                                {
                                  for(j=0;j<a;j++)
                                  {
                                  cout<<"\t "<<arr2[j][i];
                                  }
                                  cout<<"\n";
                                }
                                cout<<"\n";
                                break;  

                    case '3':   cout<<endl<<"Here is Your Transpose of The Both Matrixes: ";
                                cout<<endl<<"The Transpose of 1st Matrix:- \n\n";
                                for(i=0;i<b;i++)
                                {
                                  for(j=0;j<a;j++)
                                  {
                                  cout<<"\t "<<arr[j][i];
                                  }
                                  cout<<"\n";
                                }
                                cout<<endl<<"The Transpose of 2nd Matrix:- \n\n";
                                for(i=0;i<b;i++)
                                {
                                  for(j=0;j<a;j++)
                                  {
                                  cout<<"\t "<<arr2[j][i];
                                  }
                                  cout<<"\n";
                                }
                                cout<<"\n";
                                break;

                      default: cout<<endl<<"Invalid";
                               cout<<endl<<"Please Try Again";
                               cout<<"\n";
                               break;
                   }
                   break;

      case 'U':   cout<<endl<<"Welcome in The Upper Triangular Matrix "<<endl;
                  char choose2;
                  cout<<endl<<"If you want to Print Only 1st Matrix Then Type (1)"<<endl<<"If you want to Print Only 2nd Matrix Then Type (2)"<<endl<<"If you want to Print Both Matrixes Then Type (3)";
                  cout<<endl<<"\nEnter Your Choice: ";
                  cin>>choose2;
                  switch(choose2)
                  {
                    case '1': cout<<endl<<"Here is Your 1st Upper Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i>j)
                                  {
                                    cout<<"\t"<<"0";
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<"\n";
                              break;

                    case '2': cout<<endl<<"Here is Your 2nd Upper Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i>j)
                                  {
                                    cout<<"\t"<<"0";
                                    
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr2[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<"\n";
                              break;

                    case '3':  cout<<endl<<"Here is Your Both Upper Triangular Matrixes: ";
                               cout<<endl<<"The 1st Upper Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i>j)
                                  {
                                    cout<<"\t"<<"0";
                                    
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<endl<<"The 2nd Upper Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i>j)
                                  {
                                    cout<<"\t"<<"0";
                                    
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr2[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<"\n";
                              break;
                      
                       default: cout<<endl<<"Invalid";
                                cout<<endl<<"Please Try Again";  
                                cout<<"\n";
                                break;       
                  }
                  break;

        case 'L': cout<<endl<<"Welcome in The Lower Triangular Matrix "<<endl;
                  char choose3;
                  cout<<endl<<"If you want to Print Only 1st Matrix Then Type (1)"<<endl<<"If you want to Print Only 2nd Matrix Then Type (2)"<<endl<<"If you want to Print Both Matrixes Then Type (3)";
                  cout<<endl<<"\nEnter Your Choice: ";
                  cin>>choose3;
                  switch(choose3)
                  {
                    case '1':  cout<<endl<<"Here is Your 1st Lower Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i<j)
                                  {
                                    cout<<"\t"<<"0";
                                    
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<"\n";
                              break;
                    
                    case '2': cout<<endl<<"Here is Your 2nd Lower Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i<j)
                                  {
                                    cout<<"\t"<<"0";
                                    
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr2[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<"\n";
                              break;

                    case '3': cout<<endl<<"Here is Your Both Lower Triangular Matrixes: ";
                               cout<<endl<<"The 1st Lower Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i<j)
                                  {
                                    cout<<"\t"<<"0";
                                  
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<endl<<"The 2nd Lower Triangular Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if(i<j)
                                  {
                                    cout<<"\t"<<"0";
                                    
                                  }
                                  else
                                  {
                                    cout<<"\t"<<arr2[i][j];
                                  }
                                }
                                cout<<"\n";
                              }
                              cout<<"\n";
                              break;
                      
                       default: cout<<endl<<"Invalid";
                                cout<<endl<<"Please Try Again";  
                                cout<<"\n";
                                break;       
                  }
                  break;

       case 'X':  cout<<endl<<"Welcome in The Diagonal Matrix "<<endl;
                  char choose4;
                  cout<<endl<<"If you want to Print Only 1st Matrix Then Type (1)"<<endl<<"If you want to Print Only 2nd Matrix Then Type (2)"<<endl<<"If you want to Print Both Matrixes Then Type (3)";
                  cout<<endl<<"\nEnter Your Choice: ";
                  cin>>choose4;
                  switch(choose4)
                  {
                    case '1':  cout<<endl<<"Here is Your 1st Diagonal Matrix: \n";
                               cout<<endl<<"Left to Right Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {  
                                  cout<<arr[i][i]<<endl;
                               }
                               cout<<endl<<"Right to Left Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {
                                cout<<arr[i][a-i-1]<<endl;
                               }
                               cout<<"\n";
                               break;
                    
                    case '2':  cout<<endl<<"Here is Your 2nd Diagonal Matrix: \n";
                               cout<<endl<<"Left to Right Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {  
                                  cout<<arr2[i][i]<<endl;
                               }
                               cout<<endl<<"Right to Left Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {
                                cout<<arr2[i][a-i-1]<<endl;
                               }
                               cout<<"\n";
                               break;
                    
                    case '3': cout<<endl<<"Here is Your Both Diagonal of Matrixes\n";
                              cout<<endl<<"Here is Your 1st Diagonal Matrix: \n";
                               cout<<endl<<"Left to Right Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {  
                                  cout<<arr[i][i]<<endl;
                               }
                               cout<<endl<<"Right to Left Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {
                                  cout<<arr[i][a-i-1]<<endl;
                               }
                               cout<<endl<<"Here is Your 2nd Diagonal Matrix: \n";
                               cout<<endl<<"Left to Right Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {  
                                  cout<<arr2[i][i]<<endl;
                               }
                               cout<<endl<<"Right to Left Diagonal Elements of Matrix:\n";
                               for(i=0;i<a;i++)
                               {
                                cout<<arr2[i][a-i-1]<<endl;
                               }
                               cout<<"\n";
                               break;

                    default: cout<<endl<<"Invalid";
                             cout<<endl<<"Please Try Again";  
                             cout<<"\n";
                             break;
 
                      }
                      break;

        case 'O': cout<<endl<<"Welcome in The Tri-Diagonal Matrix "<<endl;
                  char choose5;
                  cout<<endl<<"If you want to Print Only 1st Matrix Then Type (1)"<<endl<<"If you want to Print Only 2nd Matrix Then Type (2)"<<endl<<"If you want to Print Both Matrixes Then Type (3)";
                  cout<<endl<<"\nEnter Your Choice: ";
                  cin>>choose5;
                  switch(choose5)
                  {
                    case '1': cout<<endl<<"Here is Your 1st Tri-Diagonal Matrix:- \n\n";
                               for(i=0;i<a;i++)
                                {
                                  for(j=0;j<b;j++)
                                  {
                                    if((i==j)||(i-j==-1)||(i-j==1))
                                    {
                                      cout<<arr[i][j]<<" ";
                                    }
                                    else
                                    {
                                      cout<<"0"<<" ";
                                    }
                                  }
                                    cout<<"\n";
                                }
                                cout<<"\n";
                                break;
                    
                    case '2': cout<<endl<<"Here is Your 2nd Tri-Diagonal Matrix:= \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if((i==j)||(i-j==-1)||(i-j==1))
                                  {
                                    cout<<arr2[i][j]<<" ";
                                  }
                                  else
                                  {
                                    cout<<"0"<<" ";
                                  }
                                }
                                cout<<"\n";
                                }
                                cout<<"\n";
                                break;
                    
                    case '3': cout<<endl<<"Here is Your Both Tri-Diagonal of Matrixes\n";
                              cout<<endl<<"Here is Your 1st Tri-Diagonal Matrix:- \n\n";
                              for(i=0;i<a;i++)
                                {
                                  for(j=0;j<b;j++)
                                  {
                                    if((i==j)||(i-j==-1)||(i-j==1))
                                    {
                                      cout<<arr[i][j]<<" ";
                                    }
                                    else
                                    {
                                      cout<<"0"<<" ";
                                    }
                                  }
                                    cout<<"\n";
                                }
                              cout<<endl<<"Here is Your 2nd Tri-Diagonal Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  if((i==j)||(i-j==-1)||(i-j==1))
                                  {
                                    cout<<arr2[i][j]<<" ";
                                  }
                                  else
                                  {
                                    cout<<"0"<<" ";
                                  }
                                }
                                cout<<"\n";
                                }
                                cout<<"\n";
                                break;

                    default: cout<<endl<<"Invalid";
                             cout<<endl<<"Please Try Again"; 
                             cout<<"\n"; 
                             break; 
                  }
                  break;

        case 'N': cout<<endl<<"Welcome in The Null Matrix "<<endl;
                  char choose6;
                  cout<<endl<<"If you want to Print Only 1st Matrix Then Type (1)"<<endl<<"If you want to Print Only 2nd Matrix Then Type (2)"<<endl<<"If you want to Print Both Matrixes Then Type (3)";
                  cout<<endl<<"\nEnter Your Choice: ";
                  cin>>choose6;
                  switch(choose6)
                  {
                    case '1': cout<<endl<<"Here is Your 1st Null Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  cout<<"0"<<" ";
                                }
                                  cout<<"\n";
                              }
                              cout<<"\n";
                              break;

                    case '2': cout<<endl<<"Here is Your 2nd Null Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  cout<<"0"<<" ";
                                }
                                  cout<<"\n";
                              }
                              cout<<"\n";
                              break;

                    case '3': cout<<endl<<"Here is Your Both Null Matrixes\n";
                              cout<<endl<<"Here is Your 1st Null Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  cout<<"0"<<" ";
                                }
                                  cout<<"\n";
                              }
                              cout<<endl<<"Here is Your 2nd Null Matrix:- \n\n";
                              for(i=0;i<a;i++)
                              {
                                for(j=0;j<b;j++)
                                {
                                  cout<<"0"<<" ";
                                }
                                  cout<<"\n";
                              }
                              cout<<"\n";
                              break;

                    default: cout<<endl<<"Invalid";
                             cout<<endl<<"Please Try Again";  
                             cout<<"\n";
                             break; 
                  }
                  break;

        default: cout<<endl<<"Invalid";
                 cout<<endl<<"Please Try Again";  
                 cout<<"\n";
                 break; 

    }
    return 0;
}

File: UniversalMatrix.py
def input_matrix(rows, cols, matrix_num):
    matrix = []
    print(f"\nEnter Your's Elements in The {matrix_num} Matrix:-")
    for i in range(rows):
        row = []
        for j in range(cols):
            element = int(input(f"Enter the Element in Mat[{i}][{j}]: "))
            row.append(element)
        matrix.append(row)
    return matrix

def print_matrix(matrix, title):
    print(f"\nHere is Your {title} Matrix:-\n")
    for row in matrix:
        print("\t".join(map(str, row)))
    print("\n")

def add_matrices(matrix1, matrix2):
    return [[matrix1[i][j] + matrix2[i][j] for j in range(len(matrix1[0]))] for i in range(len(matrix1))]

def subtract_matrices(matrix1, matrix2):
    return [[matrix1[i][j] - matrix2[i][j] for j in range(len(matrix1[0]))] for i in range(len(matrix1))]

def multiply_matrices(matrix1, matrix2):
    result = [[0 for _ in range(len(matrix2[0]))] for _ in range(len(matrix1))]
    for i in range(len(matrix1)):
        for j in range(len(matrix2[0])):
            for k in range(len(matrix2)):
                result[i][j] += matrix1[i][k] * matrix2[k][j]
    return result

def transpose_matrix(matrix):
    return [[matrix[j][i] for j in range(len(matrix))] for i in range(len(matrix[0]))]

def upper_triangular(matrix):
    return [[matrix[i][j] if i <= j else 0 for j in range(len(matrix[0]))] for i in range(len(matrix))]

def lower_triangular(matrix):
    return [[matrix[i][j] if i >= j else 0 for j in range(len(matrix[0]))] for i in range(len(matrix))]

def diagonal_elements(matrix):
    left_to_right = [matrix[i][i] for i in range(len(matrix))]
    right_to_left = [matrix[i][len(matrix) - i - 1] for i in range(len(matrix))]
    return left_to_right, right_to_left

def tri_diagonal(matrix):
    return [[matrix[i][j] if i == j or i == j - 1 or i == j + 1 else 0 for j in range(len(matrix[0]))] for i in range(len(matrix))]

def null_matrix(rows, cols):
    return [[0 for _ in range(cols)] for _ in range(rows)]

def main():
    print("\n----------------------------------------------Welcome in the Universal of 2D Matrix--------------------------------------------------\n")

    rows = int(input("Enter the Row: "))
    cols = int(input("Enter the Column: "))

    matrix1 = input_matrix(rows, cols, '1st')
    matrix2 = input_matrix(rows, cols, '2nd')

    print_matrix(matrix1, '1st')
    print_matrix(matrix2, '2nd')

    print("Please Choose Your Matrix Operation From there: \n")
    print("1.Addition Matrix: A\n2.Subtraction Matrix: S\n3.Multiplication Matrix: M\n4.Transpose Matrix: T\n5.Upper Triangular Matrix: U")
    print("6.Lower Triangular Matrix: L\n7.Diagonal Matrix: X\n8.Tri-Diagonal Matrix: O\n9.Null Matrix: N\n")

    operation = input("Enter The Matrix Code: ").upper()

    if operation == 'A':
        result = add_matrices(matrix1, matrix2)
        print_matrix(result, 'Addition')

    elif operation == 'S':
        result = subtract_matrices(matrix1, matrix2)
        print_matrix(result, 'Subtraction')

    elif operation == 'M':
        result = multiply_matrices(matrix1, matrix2)
        print_matrix(result, 'Multiplication')

    elif operation == 'T':
        print("If you want to Print Only 1st Matrix Then Type (1)\nIf you want to Print Only 2nd Matrix Then Type (2)\nIf you want to Print Both Matrices Then Type (3)")
        choice = input("\nEnter Your Choice: ")
        if choice == '1':
            result = transpose_matrix(matrix1)
            print_matrix(result, 'Transpose of the 1st')
        elif choice == '2':
            result = transpose_matrix(matrix2)
            print_matrix(result, 'Transpose of the 2nd')
        elif choice == '3':
            result1 = transpose_matrix(matrix1)
            result2 = transpose_matrix(matrix2)
            print_matrix(result1, 'Transpose of the 1st')
            print_matrix(result2, 'Transpose of the 2nd')
        else:
            print("Invalid choice. Please Try Again.")

    elif operation == 'U':
        print("If you want to Print Only 1st Matrix Then Type (1)\nIf you want to Print Only 2nd Matrix Then Type (2)\nIf you want to Print Both Matrices Then Type (3)")
        choice = input("\nEnter Your Choice: ")
        if choice == '1':
            result = upper_triangular(matrix1)
            print_matrix(result, '1st Upper Triangular')
        elif choice == '2':
            result = upper_triangular(matrix2)
            print_matrix(result, '2nd Upper Triangular')
        elif choice == '3':
            result1 = upper_triangular(matrix1)
            result2 = upper_triangular(matrix2)
            print_matrix(result1, '1st Upper Triangular')
            print_matrix(result2, '2nd Upper Triangular')
        else:
            print("Invalid choice. Please Try Again.")

    elif operation == 'L':
        print("If you want to Print Only 1st Matrix Then Type (1)\nIf you want to Print Only 2nd Matrix Then Type (2)\nIf you want to Print Both Matrices Then Type (3)")
        choice = input("\nEnter Your Choice: ")
        if choice == '1':
            result = lower_triangular(matrix1)
            print_matrix(result, '1st Lower Triangular')
        elif choice == '2':
            result = lower_triangular(matrix2)
            print_matrix(result, '2nd Lower Triangular')
        elif choice == '3':
            result1 = lower_triangular(matrix1)
            result2 = lower_triangular(matrix2)
            print_matrix(result1, '1st Lower Triangular')
            print_matrix(result2, '2nd Lower Triangular')
        else:
            print("Invalid choice. Please Try Again.")

    elif operation == 'X':
        print("If you want to Print Only 1st Matrix Then Type (1)\nIf you want to Print Only 2nd Matrix Then Type (2)\nIf you want to Print Both Matrices Then Type (3)")
        choice = input("\nEnter Your Choice: ")
        if choice == '1':
            left_to_right, right_to_left = diagonal_elements(matrix1)
            print(f"\nLeft to Right Diagonal Elements of 1st Matrix:\n{left_to_right}")
            print(f"\nRight to Left Diagonal Elements of 1st Matrix:\n{right_to_left}")
        elif choice == '2':
            left_to_right, right_to_left = diagonal_elements(matrix2)
            print(f"\nLeft to Right Diagonal Elements of 2nd Matrix:\n{left_to_right}")
            print(f"\nRight to Left Diagonal Elements of 2nd Matrix:\n{right_to_left}")
        elif choice == '3':
            left_to_right1, right_to_left1 = diagonal_elements(matrix1)
            left_to_right2, right_to_left2 = diagonal_elements(matrix2)
            print(f"\nLeft to Right Diagonal Elements of 1st Matrix:\n{left_to_right1}")
            print(f"\nRight to Left Diagonal Elements of 1st Matrix:\n{right_to_left1}")
            print(f"\nLeft to Right Diagonal Elements of 2nd Matrix:\n{left_to_right2}")
            print(f"\nRight to Left Diagonal Elements of 2nd Matrix:\n{right_to_left2}")
        else:
            print("Invalid choice. Please Try Again.")

    elif operation == 'O':
        print("If you want to Print Only 1st Matrix Then Type (1)\nIf you want to Print Only 2nd Matrix Then Type (2)\nIf you want to Print Both Matrices Then Type (3)")
        choice = input("\nEnter Your Choice: ")
        if choice == '1':
            result = tri_diagonal(matrix1)
            print_matrix(result, '1st Tri-Diagonal')
        elif choice == '2':
            result = tri_diagonal(matrix2)
            print_matrix(result, '2nd Tri-Diagonal')
        elif choice == '3':
            result1 = tri_diagonal(matrix1)
            result2 = tri_diagonal(matrix2)
            print_matrix(result1, '1st Tri-Diagonal')
            print_matrix(result2, '2nd Tri-Diagonal')
        else:
            print("Invalid choice. Please Try Again.")

    elif operation == 'N':
        print("If you want to Print Only 1st Matrix Then Type (1)\nIf you want to Print Only 2nd Matrix Then Type (2)\nIf you want to Print Both Matrices Then Type (3)")
        choice = input("\nEnter Your Choice: ")
        if choice == '1':
            result = null_matrix(rows, cols)
            print_matrix(result, '1st Null')
        elif choice == '2':
            result = null_matrix(rows, cols)
            print_matrix(result, '2nd Null')
        elif choice == '3':
            result1 = null_matrix(rows, cols)
            result2 = null_matrix(rows, cols)
            print_matrix(result1, '1st Null')
            print_matrix(result2, '2nd Null')
        else:
            print("Invalid choice. Please Try Again.")

    else:
        print("Invalid Matrix Code. Please Try Again.")

if __name__ == "__main__":
    main()
